src/Compilers/Core/Portable/Compilation/Expression.cs

```diff
@@ -46,7 +46,7 @@ public static ConstantValue SynthesizeNumeric(ITypeSymbol type, int value)
                 return SynthesizeNumeric(((INamedTypeSymbol)type).EnumUnderlyingType, value);
             }
 
-            return null;
+            return ConstantValue.Bad;
```

Comment:
I think this method covers all the possible types, so `Bad` should be appropriate here. I have a (possibly unrelated) question about this line though, what does it do?
```
ConstantValue.Create(1, ConstantValueTypeDiscriminator.Int32)
```

I haven't pinned down the tests that triggered the crash yet. Will do that later and update this PR.